frontend/src/components/Contexts/MissionFilterContext.tsx

```diff
@@ -41,117 +60,263 @@ const completedStatuses = [
+    const [filterError, setFilterError] = useState<string>(defaultMissionFilterInterface.filterError)
+    const [filterIsSet, setFilterIsSet] = useState<boolean>(defaultMissionFilterInterface.filterIsSet)
+    const [filterState, setFilterState] = useState<IMissionFilterContext['filterState']>(
+        defaultMissionFilterInterface.filterState
+    )
 
     const switchPage = (newPage: number) => {
         setPage(newPage)
     }
 
-    const switchMissionName = (newMissionName: string | undefined) => {
-        setMissionName(newMissionName)
-    }
-
-    const switchStatuses = (newStatuses: MissionStatus[]) => {
-        setStatuses(newStatuses)
+    const clearFilterError = () => {
+        setFilterError('')
     }
 
-    const switchRobotName = (newRobotName: string | undefined) => {
-        setRobotName(newRobotName)
-    }
-
-    const switchTagId = (newTagId: string | undefined) => {
-        setTagId(newTagId)
-    }
-
-    const switchInspectionTypes = (newInspectionTypes: InspectionType[]) => {
-        setInspectionTypes(newInspectionTypes)
-    }
-
-    const switchMinStartTime = (newMinStartTime: number | undefined) => {
-        setMinStartTime(newMinStartTime)
-    }
-
-    const switchMaxStartTime = (newMaxStartTime: number | undefined) => {
-        setMaxStartTime(newMaxStartTime)
-    }
-
-    const switchMinEndTime = (newMinEndTime: number | undefined) => {
-        setMinEndTime(newMinEndTime)
-    }
-
-    const switchMaxEndTime = (newMaxEndTime: number | undefined) => {
-        setMaxEndTime(newMaxEndTime)
-    }
+    const filterFunctions = useMemo(
+        () => ({
+            switchMissionName: (newMissionName: string | undefined) => {
+                setFilterIsSet(true)
+                setFilterState({ ...filterState, missionName: newMissionName })
+            },
+            switchStatuses: (newStatuses: MissionStatus[]) => {
+                setFilterIsSet(true)
+                setFilterState({ ...filterState, statuses: newStatuses })
+            },
+            switchRobotName: (newRobotName: string | undefined) => {
+                setFilterIsSet(true)
+                setFilterState({ ...filterState, robotName: newRobotName })
+            },
+            switchTagId: (newTagId: string | undefined) => {
+                setFilterIsSet(true)
+                setFilterState({ ...filterState, tagId: newTagId })
+            },
+            switchInspectionTypes: (newInspectionTypes: InspectionType[]) => {
+                setFilterIsSet(true)
+                setFilterState({ ...filterState, inspectionTypes: newInspectionTypes })
+            },
+            switchMinStartTime: (newMinStartTime: number | undefined) => {
+                if (
+                    filterState.minEndTime &&
+                    newMinStartTime &&
+                    filterState.maxStartTime &&
+                    (newMinStartTime > filterState.minEndTime || newMinStartTime > filterState.maxStartTime)
```

Comment:
```suggestion
            switchMinStartTime: (newMinStartTime: number | undefined) => {
                if (
                    (newMinStartTime && filterState.minEndTime && newMinStartTime > filterState.minEndTime) || (newMinStartTime && filterState.maxStartTime && newMinStartTime > filterState.maxStartTime))
```
Want this to trigger everytime newMinStartTime > filterState.minEndTime and not only when there also is a value for filterState.maxStartTime